File: python/rp_list_checks.py
#!/usr/bin/python

'''This is a python sample client program for the RackPing Monitoring API 2.0.'''

# Program: rp_list_checks.py
# Usage: ./rp_list_checks.py
# Version: 1.0
# Env: Python 2.7.9 or newer with urllib3
# Returns: exit status is non-zero on failure
# Copyright: RackPing USA 2020
# Note: For Centos6 python 2.7.9, do:
#   yum install python-pip
#   pip install requests
#   pip install urllib3
#   source ../set.sh

import json
import os
import sys
# https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
#import urllib3
#urllib3.disable_warnings()
import requests
from requests.auth import HTTPBasicAuth
from pprint import pprint

try:
   scheme   = os.environ['RP_SCHEME']
   domain   = os.environ['RP_DOMAIN']
   base_url = os.environ['RP_BASE_URL']
   timeout  = float(os.environ['RP_TIMEOUT'])
   api_key  = os.environ['RP_API_KEY']
   user     = os.environ['RP_USER']
   password = os.environ['RP_PASSWORD']
except KeyError as e:
   sys.exit('error: env not set. Please do source ../set.sh and try again: First missing key ' + e.args[0])

url = scheme + domain + base_url

### start of user settings

debug = 0

### end of user settings

if debug:
    print "url=", url

headers = {'Content-type': 'application/json',
           'App-Key': api_key
          }

msg_json_error = 'error: decoding JSON failed ...'

def output(s):
    # For a successful API call, response code will be 200 (OK) or 201 (Created)
    if debug:
       print "status=", s.status_code
    if s.ok:
        jData = json.loads(s.content)
        print json.dumps(jData)
    else:
        s.raise_for_status()
        pprint(vars(s))
    # end if
# end def output

def main():
    '''Entry point if called as an executable'''

    try:
        print "Create a request to fetch list of monitors:\n"
        myResponse = requests.get(url + "/checks", \
                                  auth=HTTPBasicAuth(user, password), \
                                  headers=headers, \
                                  timeout=timeout)
        output(myResponse)
    except requests.exceptions.RequestException as e:
        sys.exit(e)
    except ValueError as e:
        print msg_json_error, e
        sys.exit()
    # end try

# end def main

if __name__ == '__main__':
    main()
# end if

File: perl/rp_list_checks.pl
#!/usr/bin/perl

# Program: rp_list_checks.pl
# Usage: ./rp_list_checks
# Purpose: perl language sample client program for RackPing Monitoring API 2.0
# Version: 1.0
# Copyright: RackPing USA 2020
# Env: Perl5
# Returns: exit status is non-zero on failure
# Note: First set the envariables with: source ../set.sh

use strict;
use diagnostics;

use MIME::Base64 qw(encode_base64);
use LWP::UserAgent;

   use constant TIMEOUT_SEC => 5.0;

   my $url = $ENV{'RP_SCHEME'} . $ENV{'RP_DOMAIN'} . $ENV{'RP_BASE_URL'};

   my $api_key       = $ENV{'RP_API_KEY'}  // '';
   my $user          = $ENV{'RP_USER'}     // '';
   my $password      = $ENV{'RP_PASSWORD'} // '';
   my $timeout       = $ENV{'RP_TIMEOUT'}  || TIMEOUT_SEC;
   my $max_redirects = $ENV{'RP_REDIRECTS'} || 3;
   my $useragent     = $ENV{'RP_USERAGENT'} // '';
   my $DEBUG         = $ENV{'RP_DEBUG'}    || 0;

   if ($api_key eq '') {
      print "error: do source ../set.sh to set the envariables.\n";
      exit 1;
   }

### start of user settings

### end of user settings

   my @headers = (
      'Accept'         => 'application/json',
      'Accept-Charset' => 'utf-8',
      'App-key'        => "$api_key",
      'Authorization'  => 'Basic ' . encode_base64("$user:$password", ''),
      'User-Agent'     => $useragent,
   );

# Create a user agent object
   my $ua = LWP::UserAgent->new;
   $ua->from('rackping@example.com');
   $ua->timeout($timeout); # 411 - not reliable for https requests?
   $ua->max_redirect($max_redirects);

# Create a request for a list of checks
   my $req = HTTP::Request->new(GET => $url . '/checks');
   $req->header(@headers);

# Pass request to the user agent and get a response back
   my $res = $ua->request($req);

   print STDERR "info: show list of checks\n" if $DEBUG;

# Check the outcome of the response
   if ($res->is_success) {
      print $res->content, "\n";
   }
   else {
      print $res->status_line, "\n";
      exit 1;
   }

   exit 0;

